tests/interactive/scroll-view-sizing.js

```diff
@@ -32,26 +32,22 @@ const BOX_WIDTHS = [
 
 const SPACING = 10;
 
-function FlowedBoxes() {
-    this._init();
-}
+var FlowedBoxes = new Lang.Class({
+    Name: 'FlowedBoxes',
+    Extends: St.Widget,
 
-FlowedBoxes.prototype = {
-    _init: function() {
-	this.actor = new Shell.GenericContainer();
-        this.actor.connect('get-preferred-width', Lang.bind(this, this._getPreferredWidth));
-        this.actor.connect('get-preferred-height', Lang.bind(this, this._getPreferredHeight));
-        this.actor.connect('allocate', Lang.bind(this, this._allocate));
+    _init() {
+        this.parent();
 
 	for (let i = 0; i < BOX_WIDTHS.length; i++) {
 	    let child = new St.Bin({ width: BOX_WIDTHS[i], height: BOX_HEIGHT,
 	                             style: 'border: 1px solid #444444; background: #00aa44' })
-	    this.actor.add_actor(child);
+	    this.add_actor(child);
 	}
     },
 
-    _getPreferredWidth: function (actor, forHeight, alloc) {
-        let children = this.actor.get_children();
+    vfunc_get_preferred_width(forHeight) {
+        let children = this.get_children();
```

Comment:
Indentation looks off in a few places in this patch

Reply:
Fixed a few tabs that weren't removed in the lines I touched though.